examples/debugging/src/lib.rs

```diff
@@ -71,102 +80,107 @@ mod tests {
+    //             .expect("should be there")
+    //             .inputs
+    //             .first()
+    //             .expect("should be there")
+    //     };
+    //     let decoder = ABIDecoder::default();
+
+    //     {
+    //         // simple struct with a single generic parameter
+    //         let type_application = get_first_fn_argument("struct_w_generic");
+    //         let param_type = ParamType::try_from_type_application(type_application, &type_lookup)?;
+
+    //         let expected_struct = SimpleGeneric {
+    //             single_generic_param: 123u64,
+    //         };
+
+    //         assert_eq!(
+    //             format!("{expected_struct:?}"),
+    //             decoder.decode_as_debug_str(&param_type, &[0, 0, 0, 0, 0, 0, 0, 123])?
+    //         );
+    //     }
+    //     {
+    //         // struct that delegates the generic param internally
+    //         let type_application = get_first_fn_argument("struct_delegating_generic");
+    //         let param_type = ParamType::try_from_type_application(type_application, &type_lookup)?;
+
+    //         let expected_struct = PassTheGenericOn {
+    //             one: SimpleGeneric {
+    //                 single_generic_param: SizedAsciiString::<3>::try_from("abc")?,
+    //             },
+    //         };
+
+    //         assert_eq!(
+    //             format!("{expected_struct:?}"),
+    //             decoder.decode_as_debug_str(&param_type, &[97, 98, 99])?
+    //         );
+    //     }
+    //     {
+    //         // enum with generic in variant
+    //         let type_application = get_first_fn_argument("enum_w_generic");
+    //         let param_type = ParamType::try_from_type_application(type_application, &type_lookup)?;
+
+    //         let expected_enum = EnumWGeneric::B(10u64);
+
+    //         assert_eq!(
+    //             format!("{expected_enum:?}"),
+    //             decoder.decode_as_debug_str(
+    //                 &param_type,
+    //                 &[0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 10]
+    //             )?
+    //         );
+    //     }
+    //     {
+    //         // logged type
+    //         let logged_type = parsed_abi
+    //             .logged_types
+    //             .as_ref()
+    //             .expect("has logs")
+    //             .first()
+    //             .expect("has log");
+
+    //         let param_type =
+    //             ParamType::try_from_type_application(&logged_type.application, &type_lookup)?;
+
+    //         let expected_u8 = 1;
+
+    //         #[cfg(not(feature = "experimental"))]
+    //         let data = [0, 0, 0, 0, 0, 0, 0, 1];
+    //         #[cfg(feature = "experimental")]
+    //         let data = [1];
+
+    //         assert_eq!(
+    //             format!("{expected_u8}"),
+    //             decoder.decode_as_debug_str(&param_type, &data)?
+    //         );
+    //     }
+
+    //     Ok(())
+    // }
```

Comment:
I agree ☝️ 

But I'm out of context right now: why are we commenting these ones out?

Edit: Ah, is it because of what's being done down there at https://github.com/FuelLabs/fuels-rs/pull/1303/files#diff-83fcea21bf082a26a06f0abcc9b1ab9fa11993ff44258a1140396213cdafd6daR266?

Reply:
I have decided to remove it and add an issue to return the test. The problem is that the sway compiler is using to much resources when compiling this project with the experimental encoding. The issue: https://github.com/FuelLabs/fuels-rs/issues/1316